_examples/scatter/main.go

```diff
@@ -7,17 +7,23 @@ import (
 	_ "net/http/pprof"
 
 	"github.com/wcharczuk/go-chart"
+	"github.com/wcharczuk/go-chart/drawing"
 )
 
 func drawChart(res http.ResponseWriter, req *http.Request) {
+
+	viridisByY := func(xr, yr chart.Range, x, y float64) drawing.Color {
```

Comment:
i would even generalize this a bit by creating a `DotColorProviderShim` or similar that allows you to bridge calls from a `DotColorProvider` to a `ColorProvider`